src/west/manifest.py

```diff
@@ -1758,7 +1835,27 @@ def get_option(option, default=None):
             self._raw_config_group_filter = get_option('manifest.group-filter')
             self._config_path = manifest_path
 
+            def project_filter_val(configfile) -> Optional[str]:
+                return config.get('manifest.project-filter',
+                                  configfile=configfile)
+
+            # Update our project filter based on the value in all the
+            # configuration files. This allows us to progressively
+            # build up a project filter with default settings in
+            # configuration files with wider scope, refining as
+            # needed. For example, you could do '-foo,-bar' in the
+            # global config file, and then '+bar' in the local config
+            # file, to have a final filter of '-foo'.
```

Comment:
Nit: This comment is not fully correct. The correct comment would be
`.. in the local config file, to have a final filter of '-foo,-bar,+bar'.`

(The code is just appending more to the list.
`-foo,-bar,+bar != -foo` as on the left one, bar is active even if bar groups are disabled.)

Reply:
This paragraph is going away entirely since we're reversing course and just making the highest precedence one take effect